File: ployglots/mta-ployglot-pkg-app-yaml-overwrite/packages/engine-mjs/src/security.mjs
import { SecurityError } from './errors.mjs';

const BANNED_SEGMENTS = new Set(['__proto__', 'constructor', 'prototype']);

export class Security {
  static validatePath(path) {
    if (typeof path !== 'string') return;
    for (const part of path.split('.')) {
      if (BANNED_SEGMENTS.has(part)) {
        throw new SecurityError(`Forbidden path segment: ${part}`, { path, part });
      }
    }
  }
}

File: ployglots/mta-ployglot-pkg-app-yaml-overwrite/packages/engine-py/src/runtime_template_resolver/security.py
"""Path-traversal validation for {{path.to.value}} lookups."""

from __future__ import annotations

from .errors import SecurityError

_BANNED_SEGMENTS = frozenset({"__proto__", "constructor", "prototype", "__class__", "__dict__"})


class Security:
    @staticmethod
    def validate_path(path: str) -> None:
        if not isinstance(path, str):
            return
        for part in path.split("."):
            if part in _BANNED_SEGMENTS:
                raise SecurityError(
                    f"Forbidden path segment: {part}",
                    {"path": path, "part": part},
                )
